### DISCUSSION TITLE
https://imagekit.io/docs/api-reference/upload-file/upload-file-v2

### DISCUSSION THREAD

**Discussion by giscus** [imagekitio/discussions#62] [Docs]
# https://imagekit.io/docs/api-reference/upload-file/upload-file-v2

The V2 API enhances security by verifying the entire payload using JWT. This API is in beta.

ImageKit.io allows you to upload files directly from both the server and client sides. For server-side uploads, private API key authentication is used. For client-side uploads, generate a one-time `token` from your secure backend using private API. [Learn more](/docs/api-reference/upload-file/upload-file-v2#how-to-implement-secure-client-side-file-upload) about how to implement secure client-side file upload.

**File size limit** \
On the free plan, the maximum upload file sizes are 20MB for images, audio, and raw files, and 100MB for videos. On the paid plan, these limits increase to 40MB for images, audio, and raw files, and 2GB for videos. These limits can be further increased with higher-tier plans.

**Version limit** \
A file can have a maximum of 100 versions.

**Demo applications**

- A full-fledged [upload widget using Uppy](https://github.com/imagekit-samples/uppy-uploader), supporting file selections from local storage, URL, Dropbox, Google Drive, Instagram, and more.
- [Quick start guides](/docs/quick-start-guides) for various frameworks and technologies.

https://imagekit.io/docs/api-reference/upload-file/upload-file-v2

**Comment by carlosalvidrez** [comment]
This example is not working for me... I get an error stating the private key must be part of the ImageKit initialization (on the client!). 

  const imagekit = new ImageKit({
    publicKey: "your_public_api_key",
    urlEndpoint: "https://ik.imagekit.io/your_imagekit_id",
  });

**Reply by imagekitio** [reply]
You are using frontend SDK it seems. They don't support this beta API, you will have to call the endpoint directly.

**Comment by Dev25-00** [comment]
I need to check my credits or used credits, there is any request to do ? i'm using angular. thanks

**Comment by taverarene2323-glitch** [comment]
https://upload.imagekit.io/api/v1/files/upload

**Comment by taverarene2323-glitch** [comment]
This is helpful
